python/meterpreter/ext_server_stdapi.py

```diff
@@ -1663,6 +1663,44 @@ def stdapi_sys_process_memory_unlock(request, response):
         return error_result_windows(), response
     return ERROR_SUCCESS, response
 
+@register_function_if(has_windll)
+def stdapi_sys_process_memory_read(request, response):
+    ERROR_PARTIAL_COPY = 229
+    handle = packet_get_tlv(request, TLV_TYPE_HANDLE).get('value', 0)
+    base = packet_get_tlv(request, TLV_TYPE_BASE_ADDRESS).get('value', 0)
+    size = packet_get_tlv(request, TLV_TYPE_LENGTH).get('value', 0)
```

Comment:
This should fail with `ERROR_INVALID_PARAMETER` when any of these are not set as the Windows one does in [memory.c](https://github.com/rapid7/metasploit-payloads/blob/6bac251842e0140b78e6bd7e23e56688b83f34cb/c/meterpreter/source/extensions/stdapi/server/sys/process/memory.c#L98-L105).

The Windows one doesn't appear to allow a value of 0, so you can't use this to write to the NULL page, even if it were allocated. For the sake of simplicity, the Python one should also probably require a non-zero number.

For example, check with `if not (handle and base and size)` and not `if not (handle is not None and base is not None...)`.